private/dir.go

```diff
@@ -0,0 +1,55 @@
+package private
```

Comment:
I just don't like `private` as a oo like mean of "do not use this" because it's not how Go works unfortunately :( as said below, I'd better have different packages with meaningful names (maybe under `transports` also if they're related to transports